pkg/controllers/vdb/replicatedupgrade_reconciler.go

```diff
@@ -158,23 +158,35 @@ func (r *ReplicatedUpgradeReconciler) assignSubclustersToReplicaGroupA(ctx conte
 	return ctrl.Result{}, err
 }
 
-// runObjReconciler will run the object reconciler. This is used to build or
-// update any necessary objects the upgrade depends on.
-func (r *ReplicatedUpgradeReconciler) runObjReconciler(ctx context.Context) (ctrl.Result, error) {
-	rec := MakeObjReconciler(r.VRec, r.Log, r.VDB, r.PFacts, ObjReconcileModeAll)
-	return rec.Reconcile(ctx, &ctrl.Request{})
+// runObjReconcilerForMainCluster will run the object reconciler for all objects
+// that are part of the main cluster. This is used to build or update any
+// necessary objects the upgrade depends on.
+func (r *ReplicatedUpgradeReconciler) runObjReconcilerForMainCluster(ctx context.Context) (ctrl.Result, error) {
+	rec := MakeObjReconciler(r.VRec, r.Log, r.VDB, r.PFacts[vapi.MainCluster], ObjReconcileModeAll)
+	r.Manager.traceActorReconcile(rec)
+	res, err := rec.Reconcile(ctx, &ctrl.Request{})
+	r.PFacts[vapi.MainCluster].Invalidate()
+	return res, err
 }
 
 // runAddSubclusterReconciler will run the reconciler to create any necessary subclusters
 func (r *ReplicatedUpgradeReconciler) runAddSubclusterReconciler(ctx context.Context) (ctrl.Result, error) {
```

Comment:
```suggestion
func (r *ReplicatedUpgradeReconciler) runAddSubclusterReconcilerForMainCluster(ctx context.Context) (ctrl.Result, error) {
```
To be consistent. Let's update the description too.